README.md
# docSkimmer Theme

**docSkimmer** is a skimmable minimal theme for [MkDocs](https://github.com/mkdocs/mkdocs/). Supports [MkDocs versions >= 0.16.2](https://github.com/hfagerlund/mkdocs-docskimmer#version-compatibility).

<img style="max-width:100%;" alt="Screenshot of docSkimmer theme for MkDocs" src="/screenshots/mkdocs-docskimmer.png" align="center" /><br />

*Figure 1: Screenshot of docSkimmer theme for MkDocs*

## Features

* accessible;
* responsive;
* valid HTML5 + CSS3;
* no fonts (include only what you prefer using `extra_css`);
* no libs - except for a single (~2KB) module (include only what you prefer using `extra_javascript`);
* supports MkDocs configuration features such as `google_analytics`.

## Installation

* Clone docSkimmer from GitHub:

```
$ git clone https://github.com/hfagerlund/mkdocs-docskimmer.git

```

Or [download the required version](https://github.com/hfagerlund/mkdocs-docskimmer/releases) - refer to: ['Version compatibility'](https://github.com/hfagerlund/mkdocs-docskimmer#version-compatibility).

(Not on PyPI yet.)

* Copy the contents of the `mkdocs_docskimmer` directory into the MkDocs project root (ie. at the same level as the `docs` directory).

## Usage

* Modify the following settings in the 'mkdocs.yml' file:

### For docSkimmer v0.3.x, v0.4.x:
```yaml
# mkdocs.yml:

nav:
    ## add your custom pages hierarchy here

theme:
  name: null
  custom_dir: 'mkdocs_docskimmer'
  include_search_page: true
  search_index_only: false
  static_templates:
  - 404.html

plugins: ['search']

# add theme version number here
extra:
    version: 0.3.x

```

### For docSkimmer v0.2.1:
```yaml
# mkdocs.yml:

pages:
    ## add your custom pages hierarchy here

theme:
  name: null
  custom_dir: 'mkdocs_docskimmer'
  include_search_page: true
  search_index_only: false
  static_templates:
  - 404.html

plugins: ['search']

extra:
    version: 0.2.1

```

### For docSkimmer v0.2.0:
```yaml
# mkdocs.yml:

pages:
    ## add your custom pages hierarchy here

theme:
    name: null
    custom_dir: 'mkdocs_docskimmer'

static_templates:
    - 404.html

extra:
    version: 0.2.0

```

### For docSkimmer v0.1.x:
```yaml
# mkdocs.yml:

pages:
    ## add your custom pages hierarchy here

theme_dir: 'mkdocs_docskimmer'
```

## Version compatibility

| docSkimmer theme version(s) | MkDocs version(s) |
| :------: | :------: |
| 0.4.0 | 1.0.4 |
| 0.3.1 | 1.0.4<br>[Template URL filter addition](https://github.com/mkdocs/mkdocs/tree/decaf6141c90b0123a4b7f03ee839d0780570bc9) |
| 0.3.0 | [Pages refactor](https://github.com/mkdocs/mkdocs/tree/34ef3ca6d0390959080ce93a695361eea1649272) |
| 0.2.1 | 0.17.4<br>0.17.3<br>0.17.2<br>0.17.1<br>0.17.0 |
| 0.2.0 | 0.17.0 |
| 0.1.1<br>0.1.0 | 0.16.3<br>0.16.2 |

## Demo

A demo of the docSkimmer theme is available on its [documentation homepage](https://hfagerlund.github.io/mkdocs-docskimmer/).


## License
Copyright (c) 2017 Heini Fagerlund. Licensed under the [Simplified BSD License](https://github.com/hfagerlund/mkdocs-docskimmer/blob/master/LICENSE).


CHANGELOG.md
# Changelog:

- - -
## 0.4.0 - 2018-10-11

#### Changed:
* [Refactored sliding (mobile) menu panel](https://github.com/hfagerlund/mkdocs-docskimmer/issues/11)
* Updated [demo](https://hfagerlund.github.io/mkdocs-docskimmer/) (to docSkimmer v0.4.0)
* Bumped version number
- - -
## 0.3.1 - 2018-07-04

#### Changed:
* Updated [demo](https://hfagerlund.github.io/mkdocs-docskimmer/) (to docSkimmer v0.3.1)
* Bumped version number

#### Fixed:
* Compatible with [new MkDocs template URL filter](https://github.com/mkdocs/mkdocs/pull/1526/commits/68e6d1c1f9444ff0fe2a92c1eb76542330d88e4c)
- - -
## 0.3.0 - 2017-06-29

#### New features:
* Compatible with [Pages refactor of MkDocs](https://github.com/mkdocs/mkdocs/tree/34ef3ca6d0390959080ce93a695361eea1649272)

#### Changed:
* Updated [demo](https://hfagerlund.github.io/mkdocs-docskimmer/) (to docSkimmer v0.3.0)
* Bumped version number
- - -
## 0.2.1 - 2017-06-27

#### New features:
* Compatible with [MkDocs versions >= 0.17.0 (ie. v0.17.0, v0.17.1, v0.17.2, v0.17.3, v0.17.4)](https://github.com/mkdocs/mkdocs/releases)

#### Changed:
* Updated [demo](https://hfagerlund.github.io/mkdocs-docskimmer/) (to docSkimmer v0.2.1)
* Bumped version number

#### Fixed:
* Improved configuration for compatibility with multiple 0.17.x versions
* Fixed typos in demo and README links (to 'Version compatibility' section)
- - -
## 0.2.0 - 2017-06-26

#### New features:
* Compatible with [MkDocs v0.17.0](https://github.com/mkdocs/mkdocs/releases)
* Displays version number of theme (in generated source)

#### Changed:
* Modified README:
  * new [Version compatibility](https://github.com/hfagerlund/mkdocs-docskimmer#version) section
* Updated [demo](https://hfagerlund.github.io/mkdocs-docskimmer/) (to docSkimmer v0.2.0)
* Bumped version number
- - -
## 0.1.1 - 2017-06-15

#### New features:
* Launched [documentation site/demo](https://hfagerlund.github.io/mkdocs-docskimmer/)

#### Changed:
* Replaced screenshot
* Modified README:
  * minor text copy, format, spacing changes
  * updated (new) screenshot filename
* Bumped version number

#### Fixed:
* Fixed 404 error page bug (Search block URL)
* Corrected typos in README, (subnav) comment
- - -
## 0.1.0 - 2017-06-12
#### New features:
* Initial version

#### Known issues:
* Active/'current' class is not applied to parent/top-level navigation bar link when any page in that section - except for the **first** 'child'/sub-page - is active.
* Inline <code></code> snippets are currently displayed as *blocks* instead of *inline*.
* Code highlighting + language syntax labelling are not supported.
- - -
#### *Notes*: 
* All dates shown in YYYY-MM-DD format
* Using [SemVer](http://semver.org/)


mkdocs_docskimmer/js/base.js
var MenuPanel = (function() {
  'use strict';

  var menuOpenControl = document.getElementById('menu-hamburger');
  var containerPageToc = document.getElementById("page-toc");
  var menuCloseControl = document.getElementById('page-toc__closebtn');
  var pageTocLinks = document.getElementsByClassName("page-toc__link");
  var containerMainContent = document.getElementById("maincontent");

  function _bindEventListeners(){
    menuCloseControl.addEventListener('click', _menuCloseControlEventHandler);
    menuCloseControl.addEventListener('keydown', _menuCloseControlEventHandler);
    menuOpenControl.addEventListener('click', _menuOpenControlEventHandler);
    menuOpenControl.addEventListener('keydown', _menuOpenControlEventHandler);
    for (var i = 0; i < pageTocLinks.length; i++) {
      pageTocLinks[i].addEventListener('click', _menuCloseControlEventHandler);
      pageTocLinks[i].addEventListener('keydown', _menuCloseControlEventHandler);
    }
  }

  function _hideMenuOpenControl(){
    menuOpenControl.style.visibility = "hidden";
  }

  function _menuOpenControlEventHandler(event) {
    containerPageToc.style.width = "17em";
    containerMainContent.style.marginLeft = "19em";
  }

  function _menuCloseControlEventHandler(event) {
    //only respond to Enter or Space keys, or to click event
    if((event.keyCode === 13 || event.keyCode === 32) || (event.type === 'click')){
      containerPageToc.style.width = "0";
      containerMainContent.style.marginLeft= "2em";
      //check whether page-toc link was activated 
      if(event.target.href){
        var link = event.target.href,
            anchor = "#";
        if(link.indexOf(anchor) !== -1){
          //scroll activated anchor link to top of page
          window.location.href = event.target.href;
        }
      }
    }
  }
  
  return {
    init: function() {
      if(containerPageToc){
      	_bindEventListeners();
      } else {
        _hideMenuOpenControl();
      }
    } //END init
  };
})();

MenuPanel.init();


mkdocs_docskimmer/js/theme.js
/*
   docSkimmer theme v0.2.0
   License: BSD-2-Clause license (https://github.com/hfagerlund/mkdocs-docskimmer/blob/master/LICENSE)
*/

var MenuPanel = (function() {
  'use strict';

  var menuOpenControl = document.getElementById('menu-hamburger');
  var containerPageToc = document.getElementById("page-toc");
  var menuCloseControl = document.getElementById('page-toc__closebtn');
  var pageTocLinks = document.getElementsByClassName("page-toc__link");
  var containerMainContent = document.getElementById("maincontent");

  function _bindEventListeners(){
    menuCloseControl.addEventListener('click', _menuCloseControlEventHandler);
    menuCloseControl.addEventListener('keydown', _menuCloseControlEventHandler);
    menuOpenControl.addEventListener('click', _menuOpenControlEventHandler);
    menuOpenControl.addEventListener('keydown', _menuOpenControlEventHandler);
    for (var i = 0; i < pageTocLinks.length; i++) {
      pageTocLinks[i].addEventListener('click', _menuCloseControlEventHandler);
      pageTocLinks[i].addEventListener('keydown', _menuCloseControlEventHandler);
    }
  }

  function _hideMenuOpenControl(){
    menuOpenControl.style.visibility = "hidden";
  }

  function _menuOpenControlEventHandler(event) {
    containerPageToc.style.width = "17em";
    containerMainContent.style.marginLeft = "19em";
  }

  function _menuCloseControlEventHandler(event) {
    //only respond to Enter or Space keys, or to click event
    if((event.keyCode === 13 || event.keyCode === 32) || (event.type === 'click')){
      containerPageToc.style.width = "0";
      containerMainContent.style.marginLeft= "2em";
      //check whether page-toc link was activated
      if(event.target.href){
        var link = event.target.href,
            anchor = "#";
        if(link.indexOf(anchor) !== -1){
          //scroll activated anchor link to top of page
          window.location.href = event.target.href;
        }
      }
    }
  }

  return {
    init: function() {
      if(containerPageToc){
        _bindEventListeners();
      } else {
        _hideMenuOpenControl();
      }
    } //END init
  };
})();

MenuPanel.init();
